redbot/cogs/mod/mod.py

```diff
@@ -877,7 +877,15 @@ def __unload(self):
         if len(nickname) > 32:
             await ctx.send("The nickname you have provided is too long.")
         else:
-            await user.edit(reason=get_audit_reason(ctx.author, None), nick=nickname)
+            try:
+                await user.edit(reason=get_audit_reason(ctx.author, None), nick=nickname)
+            except discord.Forbidden:
```

Comment:
I'd prefer doing a hierarchy and perm check before the API call here.

The other error still needs to be caught for cases where the nick limitation is a result of discord not sharing all invalid nick parameters.